mapswipe_workers/mapswipe_workers/project_types/base/project.py

```diff
@@ -215,21 +214,14 @@ def save_to_firebase(self, project, groups, groupsOfTasks):
 
         logger.info(f"there are {len(groupsOfTasks)} groups for this project")
         c = 0
-        for group_id, tasks_list in groupsOfTasks.items():
+        for group_id in groupsOfTasks.keys():
+            tasks_list = groupsOfTasks[group_id]
             c += 1
             if self.projectType in [ProjectType.FOOTPRINT.value]:
-                # we compress tasks for footprint project type using gzip
-                json_string_tasks = (
-                    json.dumps(tasks_list).replace(" ", "").replace("\n", "")
-                )
-                compressed_tasks = gzip_str.gzip_str(json_string_tasks)
-                # we need to decode back, but only when using Python 3.6
-                # when using Python 3.7 it just works
-                # Unfortunately the docker image uses Python 3.7
-                encoded_tasks = base64.b64encode(compressed_tasks).decode("ascii")
+                compressed_tasks = gzip_str.compress_tasks(tasks_list)
                 task_upload_dict[
                     f"v2/tasks/{self.projectId}/{group_id}"
-                ] = encoded_tasks
+                ] = compressed_tasks
             else:
                 task_upload_dict[f"v2/tasks/{self.projectId}/{group_id}"] = tasks_list
```

Comment:
this last line is executed for all non-BF projects? Does this include building_footprint ones, and does it mean you're defining the tasks for them?

Reply:
Hey, so basically these lines do the following:
- if the project is of type building footprint --> first compress the tasks and then upload to firebase
- if the project is of any other type (built_area, change detection, completeness) --> directly upload to firebase, no compression needed for these

Reply:
I also added a comment about this to the code